packages/backend/src/event/services/event.service.ts

```diff
@@ -151,16 +151,68 @@ class EventService {
         return cleanedEvent;
       });
     }
-    const response = await mongoService.db
-      .collection(Collections.EVENT)
-      .insertMany(eventsToInsert as unknown as OptionalId<Event_Core[]>[]);
 
-    if (response.acknowledged && response.insertedCount !== events.length) {
-      throw error(
-        EventError.MissingGevents,
-        `Only ${response.insertedCount}/${events.length} saved`,
-      );
-    }
+    // Creates events in GCal
+    const processedEvents = await Promise.all(
+      eventsToInsert.map(async (event) => {
+        // Convert ObjectId to string if present
+        const eventToProcess = {
+          ...event,
+          _id: event._id ? event._id.toString() : undefined,
+        } as Event_Core;
+
+        const { _event, isRecurring, syncToGcal } = getCreateParams(
+          eventToProcess.user,
+          eventToProcess,
+        );
+
+        if (syncToGcal) {
```

Comment:
`createMany` wasn't adding event to Gmail, I mimic the functionality in `create`. This works for timedEvents and allDay events. Still having issues with recurrent events.